core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -245,4 +270,170 @@ public String getRequestURI() {
         public void destroy() {
         }
     };
+
+    
+    /**
+     * Gets the suggested plugin list from the update sites, falling back to a local version
+     */
+    @CheckForNull
+    public JSONArray getPlatformPluginList() {
+        Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
+        JSONArray initialPluginList = null;
+        updateSiteList: for (UpdateSite updateSite : Jenkins.getInstance().getUpdateCenter().getSiteList()) {
+            String updateCenterJsonUrl = updateSite.getUrl();
+            String suggestedPluginUrl = updateCenterJsonUrl.replace("/update-center.json", "/platform-plugins.json");
+            try {
+                URLConnection connection = ProxyConfiguration.open(new URL(suggestedPluginUrl));
+                
+                try {
+                    if(connection instanceof HttpURLConnection) {
+                        int responseCode = ((HttpURLConnection)connection).getResponseCode();
```

Comment:
@oleg-nenashev the proxy now defines a shorter global timeout, so things like this won't take forever